hiccup.ipynb

```diff
@@ -0,0 +1,137 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "gorgeous-ferry",
+   "metadata": {},
+   "source": [
+    "# The hiccups\n",
+    "\n",
+    "As a researcher I am interested in detecting hiccup, but need more date so I want to generate some time series.\n",
+    "The scenario I want to model is\n",
+    "* Hiccups should be observed for a week, except during sleeping (23:00 - 8:00).\n",
+    "* A hiccup happens once an hour at a random time\n",
+    "* Time series should have a time point every minute\n",
+    "\n",
+    "Other scenarios that would nice\n",
+    "1. Multiple hiccups occur in a bout (see https://en.wikipedia.org/wiki/Hiccup for histogram)\n",
+    "1. Person has infrequent multiple hiccups once a day\n",
+    "1. Person has higher chance of hiccup after eating (from 8:00 - 9:00, 12:00 - 13:00 and 18:00 - 19:00)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "appreciated-arkansas",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from math import floor\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "from sequgen.deterministic.triangular_peak import triangular_peak\n",
+    "from sequgen.dimension import Dimension\n",
+    "from sequgen.parameter_space import ParameterSpace\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "julian-bhutan",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "timepoints = (np.concatenate((\n",
+    "    np.arange('2020-03-01T08:00', '2020-03-01T23:00', dtype='datetime64[m]'),\n",
+    "    np.arange('2020-03-02T08:00', '2020-03-02T23:00', dtype='datetime64[m]'),\n",
+    "    np.arange('2020-03-03T08:00', '2020-03-03T23:00', dtype='datetime64[m]'),\n",
+    "    np.arange('2020-03-04T08:00', '2020-03-04T23:00', dtype='datetime64[m]'),\n",
+    "    np.arange('2020-03-05T08:00', '2020-03-05T23:00', dtype='datetime64[m]'),\n",
+    "))  - np.datetime64('2020-03-01T08:00')).astype(int)  # Convert datatimes into minutes since 2020-03-01T08:00"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "early-reduction",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "space = ParameterSpace((\n",
+    "    Dimension('placement', 0, 59),  # minute in an hour\n",
+    "    Dimension('height', 5, 10),  # severity\n",
+    "))\n",
+    "\n",
+    "hiccups = np.zeros(len(timepoints), dtype=float)\n",
+    "for day in range(5):\n",
+    "    daily_offset = day * (24 * 60)\n",
+    "    for hour in range(15):\n",
+    "        parameters = space.sample()\n",
+    "        # add offset, watchout placement should pick a value from timepoints, not an index\n",
+    "        parameters['placement'] = floor(parameters['placement']) + daily_offset + hour * 60\n",
+    "        hiccup = triangular_peak(timepoints, width_base_left=0, width_base_right=2, **parameters) # TODO unable to make peak of 1 time point\n",
+    "        hiccups += hiccup\n",
```

Comment:
```suggestion
    "        hiccups[parameters['placement']] = parameters['height']\n",
```
This change will generate a peak of width zero

Reply:
I want the package to fix it, not do a workaround here.